File: back-end/src/main/java/server/model/GrupaStudenataPredmet.java
package server.model;

import java.util.List;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class GrupaStudenataPredmet {

	@Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	@ManyToOne(optional = false)
	private GrupaStudenata grupaStudenata;
	
	@ManyToOne(optional = false)
	private Predmet predmet;

	@Column(nullable = false)
    private Boolean vidljiv = true;
}

File: front-end/src/app/models/GrupaStudenataPredmet.ts
import { GrupaStudenata } from "./GrupaStudenata";
import { Predmet } from "./Predmet";


export interface GrupaStudenataPredmet {
  id?: number;
  grupaStudenata: GrupaStudenata;  
  predmet: Predmet;                 
  vidljiv: boolean;
}
